cp23/hpi-fun2/roomcount/executables/generator.cpp

```diff
@@ -0,0 +1,119 @@
+#include <testlib.h>
+#include <bits/stdc++.h>
+
+using namespace std;
+
+
+const string_view SAMPLE1 = R"(5 8
+1 2
+1 3
+2 3
+2 4
+2 5
+3 4
+3 5
+4 5
+)";
+
+const string_view SAMPLE2 = R"(5 4
+3 1
+2 5
+4 2
+4 5
+)";
+
+void maxLengthTestcase(int maxSize) {
+  int n = maxSize;
+  int m = n - 1;
+
+  cout << n << " " << m << endl;
+
+  for (int i = 1; i <= m; i++) {
+    cout << i << " " << i + 1 << endl;
+  }
+}
+
+void maxWidthTestcase(int maxSize) {
+  int n = sqrt(2 * maxSize);
+  int m = n * (n-1) / 2;
+
+  cout << n << " " << m << endl;
+
+  for (int i = 1; i < n; i++) {
+    for (int j = i + 1; j <= n; j++) {
+      cout << i << " " << j << endl;
+    }
+  }
+}
+
+void intervalGraph(vector<int> &intervalStarts, int intervalLength, vector<pair<int, int>> &edges) {
+  sort(intervalStarts.begin(), intervalStarts.end());
+  for (int i = 0; i < intervalStarts.size(); i++) {
+    int end = intervalStarts[i] + intervalLength;
+    int iterator = i + 1;
+    while (iterator < intervalStarts.size() && intervalStarts[iterator] < end) {
+      edges.push_back({i + 1, ++iterator});
+    }
+  }
+}
+
+void randomTestcase(int intervalLength, int n, int maxStart) {
+  vector<int> intervalStarts;
+  for (int i = 0; i < n; i++) {
+    intervalStarts.push_back(rnd.next(0, maxStart));
+  }
+  vector<pair<int, int>> edges;
+  intervalGraph(intervalStarts, intervalLength, edges);
+  random_device rd;
+  mt19937 g(rd());
+  shuffle(edges.begin(), edges.end(), g);
```

Comment:
```suggestion
  shuffle(edges.begin(), edges.end());
```
Please use testlibs shuffle